### PR TITLE
Add circuit breaker factory as another variation of CB options

### PR DESCRIPTION
**Pull request by DeTeam**
This is first of a few PR's I'm making:

- [x] Add CB factory option
- [ ] Add state bag to request params (so that we can carry over some state and don't have to pass factory with every request)
- [ ] Maybe moving files around for readability purposes & adjust types *(optional)*

### CODE REVIEW

File: lib/client.ts [hunk 1]
```diff
@@ -11,6 +11,25 @@ const CircuitBreaker = require('circuit-breaker-js')
...
+  timeoutDuration?: number
+  errorThreshold?: number
+  volumeThreshold?: number
+}
+
+/**
+ * This interface defines factory function for getting a circuit breaker
+ */
+export interface CircuitBreakerFactory {
+  (params: ServiceClientRequestOptions): any
```
Comment (grassator) [inline lib/client.ts]: Shouldn't this return a circuit breaker?

**Comment by grassator** [comment]
👍

File: lib/client.ts (hunk 1 above)
Comment (DeTeam) [reply lib/client.ts]: It should :) we're missing some types (I'd do this in another PR and mb event send types to DT...)

**Comment by grassator** [comment]
👍

**Comment by DeTeam** [comment]
:+1:

**Comment by grassator** [comment]
👍

**Pull request merged**